packages/gen-ai/frontend/src/app/AIAssets/components/mcp/MCPServersToolbar.tsx

```diff
@@ -0,0 +1,79 @@
+import * as React from 'react';
+import { useNavigate } from 'react-router-dom';
+import { Button, Toolbar, ToolbarContent, ToolbarItem } from '@patternfly/react-core';
+import { PlayIcon, FilterIcon, SyncAltIcon } from '@patternfly/react-icons';
+import DashboardSearchField, {
+  SearchType,
+} from 'mod-arch-shared/dist/components/DashboardSearchField';
+import { PLAYGROUND_URL_PREFIX } from '~/app/utilities/const';
+
+interface MCPServersToolbarProps {
+  filterValue: string;
+  onFilterChange: (value: string) => void;
+  searchType: SearchType;
+  onSearchTypeChange: (type: SearchType) => void;
+  selectedCount: number;
+  selectedServerIds: string[];
+  onTryInPlayground: (serverIds: string[]) => void;
+  onRefresh?: () => void;
+}
+
+const MCPServersToolbar: React.FC<MCPServersToolbarProps> = ({
+  filterValue,
+  onFilterChange,
+  searchType,
+  onSearchTypeChange,
+  selectedCount,
+  selectedServerIds,
+  onTryInPlayground,
+  onRefresh,
+}) => {
+  const navigate = useNavigate();
+
+  const handleTryInPlayground = React.useCallback(() => {
+    // Save selections to playground and navigate
+    onTryInPlayground(selectedServerIds);
+    navigate(PLAYGROUND_URL_PREFIX);
```

Comment:
This could have been done using react-router state instead. But I guess it will need a bigger refactor of how state is being managed for the MCP use case.